src/core/ddsi/include/dds/ddsi/q_entity.h

```diff
@@ -369,6 +374,10 @@ struct proxy_writer {
   void *ddsi2direct_cbarg;
 };
 
+
+struct serdata;
+typedef int (*filter_function)(struct writer *wr, struct proxy_reader *prd, struct ddsi_serdata *serdata);
```

Comment:
The function typedefs tend to have the name of the function suffixed with ``_t``, ``_fn`` or ``_fn_t``. While one could argue that with such inconsistency there is no reason not to add another one, I think it would be better to name it ``filter_fn_t``.